src/ensembl/production/metadata/api/adaptors/genome.py

```diff
@@ -221,6 +221,32 @@ def fetch_genomes(self, genome_id=None, genome_uuid=None, genome_tag=None, organ
                 .join(OrganismGroupMember.organism_group) \
                 .filter(OrganismGroup.name.in_(group) | OrganismGroup.code.in_(group))
 
+        # genome group logic
+        if genome_group_id or genome_group_name or genome_group_type or genome_group_reference_only:
+            genome_select = genome_select.join(
+                GenomeGroupMember, Genome.genome_id == GenomeGroupMember.genome_id
+            ).join(
+                GenomeGroup, GenomeGroup.genome_group_id == GenomeGroupMember.genome_group_id
+            )
+
+            if genome_group_id:
+                genome_group_id = check_parameter(genome_group_id)
```

Comment:
I trust this is fine, but as a rule of thumb, it is inadvisable to change a parameter's value inside a function: the issues that can stem from this are pretty hard to debug.